### PR TITLE
20x performance for merkle tree lib

### PR DESCRIPTION
**Pull request by rstormsf**
Fixes #
When I tested with 30,000 leafs, it look ~ 23seconds to generate the tree. 
With this PR, it optimizes the performance in 20x.

### CODE REVIEW

File: test/helpers/merkleTree.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ class MerkleTree {
 
   bufDedup (elements) {
     return elements.filter((el, idx) => {
-      return this.bufIndexOf(el, elements) === idx;
+      return idx > 0 && !elements[idx - 1].equals(el);
```
Comment (nventuro) [inline test/helpers/merkleTree.js]:
```suggestion
      return idx === 0 || !elements[idx - 1].equals(el);
```
Unless I'm mistaken, the proposed implementation always drops the first element, as opposed to keeping it.

**Comment by nventuro** [comment]
Thanks for this @rstormsf! I'm curious, how are you using `merkleTree`? We've migrated most test helpers over to [openzeppelin-test-helpers](https://github.com/OpenZeppelin/openzeppelin-test-helpers#readme), but kept this one here, since  we weren't really sure how useful it is on its own.

Are you using `merkleTree` in your own tests, or for some other purpose? Would it be useful to you if it was included in some js utilities package?

File: test/helpers/merkleTree.js (hunk 1 above)
Comment (rstormsf) [reply test/helpers/merkleTree.js]: thanks for catching it! You are right.

**Comment by rstormsf** [comment]
@nventuro I've been using Merkle tree as ERC20 voting app: 
https://vote.vault12.com - it uses it for production
https://devpost.com/software/erc20-voting-app - hackathon

I also use it for airdrops:
https://rstormsf.github.io/slides-merkleairdrop/#/

I will deploy Pull strategy for merkle tree today at https://multisender.app

**Comment by rstormsf** [comment]
@nventuro https://twitter.com/rstormsf/status/1138703099445428224 - feel free to test this new feature on Kovan testnet

**Comment by nventuro** [comment]
Awesome @rstormsf, this is super interesting! We've been discussing the possibility of bundling small JavaScript utilities along our contracts, that help use some of them. MerkleTree, as well as signing messages, etc. would fit this intent very nicely.

**Pull request merged**